File: src/ast/DeleteExpression.java
package ast;

import java.io.StringWriter;

public class DeleteExpression extends ParentNode {
    private boolean _isGlobal;
    private boolean _isArrayExpression;

    public DeleteExpression(BaseNode child, boolean isGlobal, boolean isArrayExpression) {
        super(NodeType.DeleteExpression, child);
        _isGlobal = isGlobal;
        _isArrayExpression = isArrayExpression;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        if (_isGlobal) {
            writer.write("::");
        }

        writer.write("delete");

        if (_isArrayExpression) {
            writer.write("[] ");
        }

        child.Print(writer);
    }
}

File: src/de/fearlesstobi/demangler/ast/DeleteExpression.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class DeleteExpression(child: BaseNode?, private val isGlobal: Boolean, private val isArrayExpression: Boolean) : ParentNode(NodeType.DeleteExpression, child) {
    public override fun printLeft(writer: StringWriter) {
        if (isGlobal) {
            writer.write("::")
        }
        writer.write("delete")
        if (isArrayExpression) {
            writer.write("[] ")
        }
        child!!.print(writer)
    }

}